core/include/kernel/mutex.h

```diff
@@ -95,10 +95,19 @@ void condvar_broadcast_debug(struct condvar *cv, const char *fname, int lineno);
 void condvar_wait_debug(struct condvar *cv, struct mutex *m,
 			const char *fname, int lineno);
 #define condvar_wait(cv, m) condvar_wait_debug((cv), (m), __FILE__, __LINE__)
+
+TEE_Result condvar_wait_timeout_debug(struct condvar *cv, struct mutex *m,
+				      uint32_t timeout_ms, const char *fname,
+				      int lineno);
+#define condvar_wait_timeout(cv, m, timeout_ms) \
+			     condvar_wait_timeout_debug((cv), (m), \
+			     (timeout_ms), __FILE__, __LINE__)
 #else
 void condvar_signal(struct condvar *cv);
 void condvar_broadcast(struct condvar *cv);
 void condvar_wait(struct condvar *cv, struct mutex *m);
+TEE_Result condvar_wait_timeout(struct condvar *cv, struct mutex *m,
```

Comment:
I suggest to add an inline comment stating something like:
`/* Always return TEE_SUCCESS when @timeout_ms is 0 */`
It would clarify why some callers of to `wq_wait_final()` do drop its return value.

Ditto for `condvar_wait_timeout_debug()`  at line 99.

Reply:
Ok.